tests/src/test/scala/cats/tests/RegressionSuite.scala

```diff
@@ -103,7 +103,7 @@ class RegressionSuite extends CatsSuite with ScalaVersionSpecificRegressionSuite
 
   test("#500: foldMap - traverse consistency") {
     assert(
-      List(1, 2, 3).traverse(i => Const.of[List[Int]](List(i))).getConst == List(1, 2, 3).foldMap(List(_))
+      List(1, 2, 3).traverse(i => Const[List[Int], List[Int]](List(i))).getConst == List(1, 2, 3).foldMap(List(_))
```

Comment:
Dotty was inferring `Int => Int`, I'll try to reach out to the dotty team on how to get this working